src/pages/home/ReportScreen.js

```diff
@@ -1,54 +1,108 @@
+    }).isRequired,
 
-const defaultProps = {
-    currentlyViewedReportID: '0',
+    /* Navigation route context info provided by react navigation */
+    route: PropTypes.shape({
+        /* Route specific parameters used on this screen */
+        params: PropTypes.shape({
+            /* The ID of the report this screen should display */
+            reportID: PropTypes.string,
+        }).isRequired,
+    }).isRequired,
 };
 
-const ReportScreen = (props) => {
-    const activeReportID = parseInt(props.currentlyViewedReportID, 10);
-    if (!activeReportID) {
-        return null;
-    }
-
-    return (
-        <ScreenWrapper
-            style={[
-                styles.appContent,
-                styles.flex1,
-                styles.flexColumn,
-            ]}
-        >
-            <HeaderView
-                reportID={activeReportID}
-                onNavigationMenuButtonClicked={() => Navigation.navigate(ROUTES.HOME)}
-            />
-            <View style={[styles.dFlex, styles.flex1]}>
-                <ReportView reportID={activeReportID} />
-            </View>
-        </ScreenWrapper>
-    );
-};
+class ReportScreen extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isLoading: true,
+        };
+    }
+
+    componentDidMount() {
+        this.prepareTransition();
+    }
+
+    componentDidUpdate(prevProps) {
+        const reportChanged = this.props.route.params.reportID !== prevProps.route.params.reportID;
+
+        if (reportChanged) {
+            this.prepareTransition();
+        }
+    }
+
+    componentWillUnmount() {
+        clearTimeout(this.loadingTimerId);
+    }
+
+    /**
+     * Get the currently viewed report ID as number
+     *
+     * @returns {Number}
+     */
+    getReportID() {
+        const params = this.props.route.params;
+        return Number.parseInt(params.reportID, 10);
+    }
+
+    /**
+     * When reports change there's a brief time content is not ready to be displayed
+     *
+     * @returns {Boolean}
+     */
+    isReadyToDisplayReport() {
+        return !this.state.isLoading && Boolean(this.getReportID());
+    }
+
+    /**
+     * Configures a small loading transition of fixed time and proceeds with rendering available data
+     */
+    prepareTransition() {
+        if (!this.getReportID()) { return; }
+
+        this.setState({isLoading: true});
+
+        clearTimeout(this.loadingTimerId);
+        this.loadingTimerId = setTimeout(() => this.setState({isLoading: false}), 300);
+    }
+
+    render() {
+        return (
+            <ScreenWrapper
+                style={[
+                    styles.appContent,
+                    styles.flex1,
+                    styles.flexColumn,
+                ]}
+            >
+                <HeaderView
+                    reportID={this.getReportID()}
+                    onNavigationMenuButtonClicked={this.props.navigation.openDrawer}
```

Comment:
More of a question, but is there some reason this is preferable to doing what we had before and using the programmatic navigation?

Reply:
Not really - it's just every `Screen` component receives a handle to the navigation that it belongs to

I prefer to use whatever react native navigation specifically passes to this screen. Also tested if this somehow affects the mWeb routing issue - it doesn't 
I'll revert it so it doesn't get in the way if you plan to do some custom/additional stuff in `Navigation.openDrawer()`